pkg/controller/inferenceservice/controller_test.go

```diff
@@ -1136,3 +1136,184 @@ func TestInferenceServiceWithTransformer(t *testing.T) {
+		Namespace: namespace}
+	var transformer = &kfserving.InferenceService{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: namespace,
+		},
+		Spec: kfserving.InferenceServiceSpec{
+			Default: kfserving.EndpointSpec{
+				Predictor: kfserving.PredictorSpec{
+					DeploymentSpec: kfserving.DeploymentSpec{
+						MinReplicas: 1,
+						MaxReplicas: 3,
+					},
+					Tensorflow: &kfserving.TensorflowSpec{
+						StorageURI:     "s3://test/mnist/export",
+						RuntimeVersion: "1.13.0",
+					},
+				},
+				Transformer: &kfserving.TransformerSpec{
+					DeploymentSpec: kfserving.DeploymentSpec{
+						MinReplicas: 1,
+						MaxReplicas: 3,
+					},
+					Custom: &kfserving.CustomSpec{
+						Container: v1.Container{
+							Image: "transformer:v1",
+						},
+					},
+				},
+			},
+			CanaryTrafficPercent: 20,
+			Canary: &kfserving.EndpointSpec{
+				Predictor: kfserving.PredictorSpec{
+					DeploymentSpec: kfserving.DeploymentSpec{
+						MinReplicas: 1,
+						MaxReplicas: 3,
+					},
+					Tensorflow: &kfserving.TensorflowSpec{
+						StorageURI:     "s3://test/mnist-2/export",
+						RuntimeVersion: "1.13.0",
+					},
+				},
+				Transformer: &kfserving.TransformerSpec{
+					DeploymentSpec: kfserving.DeploymentSpec{
+						MinReplicas: 1,
+						MaxReplicas: 3,
+					},
+					Custom: &kfserving.CustomSpec{
+						Container: v1.Container{
+							Image: "transformer:v2",
+						},
+					},
+				},
+			},
+		},
+		Status: kfserving.InferenceServiceStatus{
+			URL: serviceName + ".svc.cluster.local",
+			Default: &kfserving.ComponentStatusMap{
+				constants.Predictor: &kfserving.StatusConfigurationSpec{
+					Name: "revision-v1",
+				},
+			},
+		},
+	}
+
+	g := gomega.NewGomegaWithT(t)
+
+	mgr, err := manager.New(cfg, manager.Options{})
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	c = mgr.GetClient()
+
+	recFn, requests := SetupTestReconcile(newReconciler(mgr))
+	g.Expect(add(mgr, recFn)).NotTo(gomega.HaveOccurred())
+
+	stopMgr, mgrStopped := StartTestManager(mgr, g)
+
+	defer func() {
+		close(stopMgr)
+		mgrStopped.Wait()
+	}()
+
+	// Create configmap
+	var configMap = &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.InferenceServiceConfigMapName,
+			Namespace: constants.KFServingNamespace,
+		},
+		Data: configs,
+	}
+	g.Expect(c.Create(context.TODO(), configMap)).NotTo(gomega.HaveOccurred())
+	defer c.Delete(context.TODO(), configMap)
+
+	// Create the InferenceService object and expect the Reconcile and knative service to be created
+	instance := transformer.DeepCopy()
+	g.Expect(c.Create(context.TODO(), instance)).NotTo(gomega.HaveOccurred())
+	defer c.Delete(context.TODO(), instance)
+	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedRequest)))
+
+	defaultPredictorService := &knservingv1alpha1.Service{}
+	g.Eventually(func() error { return c.Get(context.TODO(), defaultPredictor, defaultPredictorService) }, timeout).
```

Comment:
Would it be cleaner to loop over these 4 repeats?

Reply:
just copy these lines from other tests. I think we can just keep it as it is now since it is used for test.